### ISSUE TITLE
Markdown: Preserve double-space line break syntax

### ISSUE THREAD

**Issue by infotexture** [prettier/prettier#3319]
In the original [Markdown syntax](https://daringfireball.net/projects/markdown/syntax#p) description, two spaces at the end of a line create a line break: 

> When you do want to insert a `<br />` break tag using Markdown, you end a line with two or more spaces, then type return.

This syntax is also supported in the [CommonMark spec](http://spec.commonmark.org/0.28/#hard-line-breaks):

> A line break … that is preceded by two or more spaces and does not occur at the end of a block is parsed as a hard line break

Prettier currently replaces this with CommonMark's single-backslash notation _(which works fine in some places like GitHub issue comments)_, but this syntax is **not** supported by `kramdown`, the default Markdown parser used by GitHub Pages, so the lines are joined and the backslash characters are visible in output when prettied Markdown is rendered via GitHub Pages.

The [kramdown syntax](https://kramdown.gettalong.org/syntax.html#paragraphs) requires **two** backslash characters at the end of the line for line breaks, but this renders elsewhere as two visible backslashes rather than a line break: `\\`.

**Prettier 1.8.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAqABK1AjCAbAE3WwFc884Z1o4t10AdKe+vASwGcrJY2oSAhjDbRqUOEwxZchYmQpUYAdwh1GzFuy7oew-kJHNlEJiAA0ICAAdh0DslACuyAGYC8HOBewAnAWABrSgBlK38+AHNkGB8SLxACCDBXd08LPk8fGAAFPwiAWwEUj3iAKw4ADwAhP0CQgXy4ABk+OGK0yxIYKy6AJnb4sJ9M5BBCnwDEpShzECsfPhgAdTYCGAALZAAOAAYLeYhPJb8raNj4nzgARxI2S9yBAqKkNxKLT3y2M7j3yIoARRIEHgAwsMAE2BWa02SF6YL8bHYUAiAGEIPlCqMoDRZiRPAAVCH2F6pOAAXzJQA)
```sh
--parser markdown
```

**Input:**
```markdown
* **bold bullet one**  
    list continuation one
* **bold bullet two**  
    list continuation two

```

**Output:**
```markdown
* **bold bullet one**\
   list continuation one
* **bold bullet two**\
   list continuation two

```

**Expected behavior:**

The double-space line breaks should be preserved for maximum source compatibility with a broad range of Markdown implementations.

**Comment by azz** [comment]
Trailing whitespace affecting output is... Unpleasant. But, alas, it is a bug.

**Comment by ikatyang** [comment]
For reference, we followed the CommonMark spec, which is [not 100% compatible with Kramdown](https://github.com/gettalong/kramdown/issues/404#issuecomment-275213633).

**Comment by infotexture** [comment]
👍 Looks like this was fixed for 1.9.

Might be worth tagging with the milestone for future reference so people know which versions of Prettier will support  the `··\n` syntax.

**Comment by ikatyang** [comment]
Oops, missed this one.
